Userland/Services/WebContent/ConnectionFromClient.h

```diff
@@ -127,8 +128,9 @@ class ConnectionFromClient final
 
     HashMap<i32, NonnullRefPtr<Gfx::Bitmap>> m_backing_stores;
 
-    WeakPtr<JS::Realm> m_realm;
-    OwnPtr<WebContentConsoleClient> m_console_client;
+    HashMap<Web::DOM::Document*, OwnPtr<WebContentConsoleClient>> m_console_clients;
+    WebContentConsoleClient* m_top_level_document_console_client;
```

Comment:
What's the ownership model for this random raw pointer? :thinking:

Reply:
is is pointing to `WebContentConsoleClient` hold by `m_console_clients` that contains console client for all documents (including documents of iframes).

it is assigned non-null value in `initialize_js_console` when it is called for document corresponding to top-level browsing context.
is is nulled in `destroy_js_console` just before it is removed from `m_console_clients`.

Reply:
Instead of magically nulling it when the corresponding `WebContentConsoleClient` is destroyed, can we make it a `WeakPtr` and get the same behavior for free?

Reply:
Also, it doesn't seem valid for `m_console_clients` to contain null `WebContentConsoleClient` pointers, so it should probably have `NonnullOwnPtr` as the value type rather than `OwnPtr` :)

Reply:
yes, that all makes sense. I made these changes. thank you!